six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -43,9 +43,20 @@ NITFWriteControl::NITFWriteControl(mem::SharedPtr<Container> container)
 }
 
 NITFWriteControl::NITFWriteControl(const six::Options& options,
-                                   mem::SharedPtr<Container> container)
+                                   mem::SharedPtr<Container> container,
+                                   const XMLControlRegistry* xmlRegistry)
 {
     mNITFHeaderCreator.reset(new six::NITFHeaderCreator(options, container));
+    if (xmlRegistry)
+    {
+        setXMLControlRegistry(xmlRegistry);
```

Comment:
Avoid calling virtual functions in constructors.  This'll actually do what you want in this case, but for code clarity add a private non-virtual `setXMLControlRegistryImpl()`, call that one here, and update `NITFWriteControl::setXMLControlRegistry()` to turn around and call `setXMLControlRegistryImpl()`

Reply:
Except shoot, that turns around and calls `WriteControl::setXMLControlRegistry()` which is virtual too... so I guess add a protected `WriteControl::setXMLControlRegistryImpl()` and have `NITFWriteControl::setXMLControlRegistryImpl()` call that one.

Reply:
I've updated the ctor, and removed the non-const `getOptions()`